sympy/ntheory/residue_ntheory.py

```diff
@@ -1300,3 +1300,86 @@ def discrete_log(n, a, b, order=None, prime_order=None):
         return _discrete_log_pollard_rho(n, a, b, order)
 
     return _discrete_log_pohlig_hellman(n, a, b, order)
+
+
+
+
+def _gcd_extended(a, m):
+    """This function calculates the x and y values satisfying the equation using
+    the extended Euclidean algorithm.
+
+    ``a x + m y = gcd(a, m)``
+
+    """
+    if a == 0:
+        return 0, 1
+    else:
+        x, y = _gcd_extended(m % a, a)
+        return y - (m // a) * x , x
+
+def _lin_congurence(a, c, m):
+    """Solve the linear congurence ``a x = c mod m
+    """
+    gcd = igcd(a, m)
+    if c % gcd != 0:
+        return []
+    u, _ = _gcd_extended(a, m)
+    x = (u * c) // gcd
+    r = []
+    for t in range(0, gcd):
+        r.append(x + (m // gcd) * t)
+    return r
+
+
+
+def quadratic_congurence(a, b, c, p):
+    """
+    Find the solutions to ``a x**2 + b x + c = 0 mod p
+    a : integer
+    b : integer
+    c : integer
+    p : positive integer
+    all_roots : if False returns the smallest root, else the list of roots
+    """
+    a = as_int(a)
+    b = as_int(b)
+    c = as_int(c)
+    p = as_int(p)
+    a = a % p
+    b = b % p
+    c = (c % p)
+
+    if a == 0:
```

Comment:
This is the linear case. It can be solved by `polys.galoistools.linear_congruence`.